src/libraries/Microsoft.Extensions.Configuration.FileExtensions/src/FileConfigurationProvider.cs

```diff
@@ -53,12 +53,12 @@ public override string ToString()
 
         private void Load(bool reload)
         {
-            IFileInfo file = Source.FileProvider?.GetFileInfo(Source.Path);
+            IFileInfo? file = Source.FileProvider?.GetFileInfo(Source.Path ?? string.Empty);
```

Comment:
Why is this using a different strategy than the above change? Here we are checking for null and passing in empty. Above we are suppressing the `null` warnings.

Reply:
Great question! 
- [`GetFileInfo` actually allows nulls](https://github.dev/dotnet/runtime/blob/dc0a7f93dc6a49016c7fa7ce18e09d285f066368/src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFileProvider.cs#L265) issue is that it needs to return `NotFoundFileInfo` if file not found (which would be the case for `null`), and since `NotFoundFileInfo` is inherited from `IFileInfo`, it's `Name` cannot be `null`.
So, TLDR is `?? string.Empty` is needed so the returned `IFileInfo` would have a non-null `Name`.
This is a mistake on my part. `IFileInfo.Name` should be nullable. Updated the PR. Not sure whether `GetFileInfo.subpath` should be marked as nullable though. cc: @krwq
- As for why null is ignored in `Watch`. It also allows `null`, but after [this discussion](https://github.com/dotnet/runtime/pull/57409#discussion_r689318040) it was decided to mark it as non-nullable.

Reply:
> IFileInfo.Name should be nullable.

I don't think this is correct. If the only use case for a null `Name` on an `IFileInfo` is for the `NotFoundFileInfo`, can we just change `NotFoundFileInfo` to check for null and use `string.Empty` instead? That way all the consumers of `IFileInfo.Name` don't need to check for null.

Reply:
I thought that nullable name might be useful for some other implementations (not ours).
Reverted the changes and updated `NotFoundFileInfo` to set empty file name instead of null.